packages/geo/src/quad.key.ts

```diff
@@ -56,6 +61,45 @@ export const QuadKey = {
         return tileToBBOX(quadkeyToTile(quadKey));
     },
 
+    /**
+     * Convert a quadkey to a XYZ Tile location
+     * @param quadKey quadkey to convert
+     */
+    toTile(quadKey: string): Tile {
+        let x = 0;
+        let y = 0;
+        const z = quadKey.length;
+
+        for (let i = z; i > 0; i--) {
+            const mask = 1 << (i - 1);
+            const q = quadKey.charCodeAt(z - i);
+            if (q === CHAR_1) x |= mask;
+            if (q === CHAR_2) y |= mask;
+            if (q === CHAR_3) {
+                x |= mask;
+                y |= mask;
+            }
+        }
+        return { x, y, z };
+    },
+
+    /**
+     * Convert a tile location to a quadkey
+     * @param tile tile to covert
+     */
+    fromTile(tile: Tile): string {
+        const { x, y, z } = tile;
+        let quadKey = '';
+        for (let zI = z; zI > 0; zI--) {
+            let b = 0;
+            const mask = 1 << (zI - 1);
+            if ((x & mask) !== 0) b++;
+            if ((y & mask) !== 0) b += 2;
+            quadKey += b.toString();
+        }
+        return quadKey;
+    },
```

Comment:
I do this slightly differently

```typescript
function xyzToQuadKey(x: number, y: number, z: number): string {
    let level = 1 << z;
    if (x < 0 || x > level || y < 0 || y > level) return '';

    let qk = '';
    let hl = 0;
    let mask = 0;
    while (level > 1) {
        hl = level >> 1;
        mask = hl - 1;

        if (x < hl) {
            qk += y < hl ? '2' : '0';
        } else {
            qk += y < hl ? '3' : '1';
        }
        level = hl;
        x &= mask;
        y &= mask;
    }
    return qk;
}
```

Reply:
This logic is flawed, when running x:0,y:0,z:30, I get `2222222222...` when you should be getting `000000`

interestingly it is slightly faster though, however switching from a `b.toSTring()` to a `String.fromCharCode()` makes them about the same speed but the PR one is giving the correct answer